### PR TITLE
[CELEBORN-1296] Introduce celeborn.dynamicConfig.store.fs.path config to configure the path of dynamic config file for fs store backend

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Introduce `celeborn.dynamicConfig.store.fs.path` config to configure the path of dynamic config file for fs store backend.

### Why are the changes needed?

`FsConfigServiceImpl` uses `celeborn.quota.configuration.path` to configure the path of dynamic config file for fs store backend at present. The path of dynamic config file should be introduced with `celeborn.dynamicConfig.store.fs.path` instead of quota configuration path.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No.

### CODE REVIEW

**Comment by SteNicholas** [comment]
cc @RexXiong, @AngersZhuuuu. BTW, I will create another pull request to deprecate `celeborn.quota.configuration.path` as FOLLOWUP of CELEBORN-1239.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -4458,6 +4459,14 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("120s")
 
+  val DYNAMIC_CONFIG_STORE_FS_PATH: OptionalConfigEntry[String] =
+    buildConf("celeborn.dynamicConfig.store.fs.path")
+      .categories("master", "worker")
+      .version("0.5.0")
+      .doc("The path of dynamic config file for fs store backend.")
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4466]: Should we explain that if we didn't set it, will use  `$CELEBORN_CONF/dynamicConfig.yaml`?

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java [hunk 2]
```diff
@@ -87,14 +87,14 @@ private Map<String, String> getConfigs(Map<String, Object> configMap) {
 
   private File getConfigFile(Map<String, String> env) throws IOException {
```
Comment (AngersZhuuuu) [inline service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:88]: What if we want set quota in a separated file?

**Review by AngersZhuuuu — APPROVED** [review]

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java (hunk 2 above)
Comment (SteNicholas) [reply service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:88]: @AngersZhuuuu, why does the user set quota config in a separated file? IMO, the user always configures the dynamic config in one config file for unified management configuration.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (RexXiong) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4466]: +1

**Review by RexXiong — APPROVED** [review]
LGTM, thanks!

File: master/src/test/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManagerSuite.scala [hunk 3]
```diff
@@ -45,7 +45,7 @@ class QuotaManagerSuite extends AnyFunSuite
     val conf = new CelebornConf()
     conf.set(CelebornConf.DYNAMIC_CONFIG_STORE_BACKEND, "FS")
     conf.set(
-      CelebornConf.QUOTA_CONFIGURATION_PATH.key,
```
Comment (AngersZhuuuu) [inline master/src/test/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManagerSuite.scala:48]: Can remove this conf related code.

**Review by AngersZhuuuu — APPROVED** [review]

**Comment by SteNicholas** [comment]
LGTM. Mreged into main(v0.5.0).

**Pull request closed without merging**